aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -338,10 +339,19 @@ void gemm(
               &beta_,
               float_v.data(), &ldc_);
       at::parallel_for(0, c_size, 1, [&](int64_t begin, int64_t end) {
-        for (const auto i : c10::irange(begin, end)) {
-          *(c++) = c10::convert<at::BFloat16>(float_v[i]);
+        int64_t i = begin;
+        //Vectorized Loop
+        for (; i + c_size <= end; i += c_size) {
```

Comment:
This doesn't make sense, it will only ever take at most 1 trip since `c_size` is the upper bound for the loop.
```suggestion
        for (; i + c_size <= end; i += Vectorized<float>::size()) {
```